sklearn/utils/estimator_checks.py

```diff
@@ -404,6 +419,7 @@ def check_sample_weights_pandas_series(name, estimator_orig):
         try:
             import pandas as pd
             X = pd.DataFrame([[1, 1], [1, 2], [1, 3], [2, 1], [2, 2], [2, 3]])
+            X = gram_matrix_if_pairwise(X, estimator_orig)
```

Comment:
X output here will still not be a DataFrame, will it? There's not much point in doing this unless we make the gram matrix a DataFrame, which we might as well do even if it's a bit of a weird use of a DataFrame